contracts/base/dodo-base/DodoV1SingleLPStrategy.sol

```diff
@@ -0,0 +1,402 @@
+     *      useful in case of emergency exits.
+     */
+    function setSell(bool _sell) public onlyGovernance {
+        _setSell(_sell);
+    }
+
+    /**
+     * @dev Set the minimum amount of DODO needed to trigger a sale (for a liquidation).
+     */
+    function setSellFloor(uint256 _sellFloor) public onlyGovernance {
+        setSellFloor(_sellFloor);
+    }
+
+    function setLiquidationPath(address _token, address[] memory _route)
+        public
+        onlyGovernance
+    {
+        uniswapRoutes[_token] = _route;
+    }
+
+    function finalizeUpgrade() external onlyGovernance {
+        _finalizeUpgrade();
+
+        // Reset the liquidation paths - they need to be reset manually
+        uniswapRoutes[IDodoLpToken(underlying()).originToken()] = new address[](
+            0
+        );
+    }
+
+    // ---------------- Internal methods ----------------
+
+    function rewardPoolBalance() internal view returns (uint256 balance) {
+        (balance, ) = IDodoMine(rewardPool()).userInfo(poolId(), address(this));
+    }
+
+    function claimRewards() internal {
+        IDodoMine(rewardPool()).claim(underlying());
+    }
+
+    function enterRewardPool() internal {
+        uint256 currentBalance = IERC20(underlying()).balanceOf(address(this));
+        IERC20(underlying()).safeApprove(rewardPool(), 0);
+        IERC20(underlying()).safeApprove(rewardPool(), currentBalance);
+        IDodoMine(rewardPool()).deposit(underlying(), currentBalance);
+    }
+
+    function exitRewardPool() internal {
+        uint256 balance = rewardPoolBalance();
+        if (balance != 0) {
+            IDodoMine(rewardPool()).withdraw(underlying(), balance);
+        }
+    }
+
+    function emergencyExitRewardPool() internal {
+        uint256 balance = rewardPoolBalance();
+        if (balance != 0) {
+            IDodoMine(rewardPool()).emergencyWithdraw(underlying());
+        }
+    }
+
+    /**
+     * @dev Stakes everything the strategy holds into the reward pool.
+     */
+    function investAllUnderlying() internal onlyNotPausedInvesting {
+        // This check is needed for avoiding reverts in case we stake 0
+        if (IERC20(underlying()).balanceOf(address(this)) > 0) {
+            enterRewardPool();
+        }
+    }
+
+    // We assume that all trading can be done on Uniswap
+    function _liquidateReward() internal {
+        uint256 rewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+
+        // Profits can be disabled for possible simplified and rapid exit
+        if (!sell() || rewardBalance < sellFloor()) {
+            emit ProfitsNotCollected(sell(), rewardBalance < sellFloor());
+            return;
+        }
+
+        notifyProfitInRewardToken(rewardBalance);
+        uint256 remainingRewardBalance =
+            IERC20(rewardToken()).balanceOf(address(this));
+
+        if (remainingRewardBalance == 0) {
+            return;
+        }
+
+        // TODO: Should we liquidate DODO to USDT on DODO first
```

Comment:
This might be a good idea. Best is to do this swap before the `notifyProfitInRewardToken()`. Then USDT will be the `rewardToken` for the strategy and you notify the USDT balance. It will be better for the buyback returns.